### DISCUSSION TITLE
How can I get the response from network tab in selenide?

### DISCUSSION THREAD

**Discussion by aravindraveendran7** [selenide/selenide#2398] [Q&A] [answered]
Hello There,

I have a requirement where I need to get a response from the browser network tab to proceed further, as I don't have any APIs to get this response otherwise. I tried the below step:
**com.browserup.bup.BrowserUpProxy proxy = WebDriverRunner.getSelenideProxy().getProxy();**
 However, ended up in error as I am unable to import the pkg. Can somebody please guide me on this and set this up step by step?

**Comment by asolntsev** [comment]
@aravindraveendran7 Are you using dependency `com.codeborne:selenide` or `com.codeborne:selenide-proxy`?

**Reply by aravindraveendran7** [reply]
I am using ` com.codeborne:selenide`

**Reply by asolntsev** [reply, marked as answer]
That's the reason.
You need to replace "com.codeborne:selenide" by "com.codeborne:selenide-proxy".

And I believe your error message says exactly this. ;)

**Reply by aravindraveendran7** [reply]
yes, this was the cause of the error. Thank you.
I tried both ways as per the [doc:](https://selenide.org/2019/12/18/advent-calendar-network-logs-with-proxy/) with the latest changes as below. However, this gives me a null pointer exception at the very first line of getting selenideproxy() method. Is there something am missing here?

 1. BrowserUpProxy proxy = WebDriverRunner.getSelenideProxy().responseFilter("/accounts");
        proxy.setHarCaptureTypes(CaptureType.getAllContentCaptureTypes());
        proxy.enableHarCaptureTypes(CaptureType.RESPONSE_CONTENT);
        proxy.newHar("accounts");
        List<HarEntry> requests = proxy.getHar().getLog().getEntries();

2. BrowserUpProxy proxy = WebDriverRunner.getSelenideProxy().getProxy()
        proxy.setHarCaptureTypes(CaptureType.getAllContentCaptureTypes());
        proxy.enableHarCaptureTypes(CaptureType.RESPONSE_CONTENT);
        proxy.newHar("accounts");
        List<HarEntry> requests = proxy.getHar().getLog().getEntries();

**Reply by asolntsev** [reply]
Yes, method `WebDriverRunner.getSelenideProxy()` returns **null** if _webdriver is not started yet_. 
You need to first call `open` method, and only then add proxy filters.

**Reply by asolntsev** [reply]
@aravindraveendran7 But still, I will improve the error message. Now you will see a clear message saying what happened instead of just NullPointerException.

See https://github.com/selenide/selenide/pull/2406

**Reply by aravindraveendran7** [reply]
@asolntsev Thank you. I am using selenium to initiate the driver as my organization framework uses it. So what I did is I just passed the selenium driver to WebDriverRunner.setDriver(driver). When I use, **WebDriverRunner.getSelenideProxy().getProxy()**, how can I pass the earlier set driver to this? Please advise.

**Reply by asolntsev** [reply]
> I am using selenium to initiate the driver as my organization framework uses it.

@aravindraveendran7 But does your organization framework initialize webdriver **with proxy**?
1. If so, you need to replace method call `WebDriverRunner.setDriver(driver)` by `WebDriverRunner.setDriver(driver, proxy)`.
2. If don't open webdriver with proxy, how could you download files via proxy?

**Comment by aravindraveendran7** [comment]
> 1. If so, you need to replace method call `WebDriverRunner.setDriver(driver)` by `WebDriverRunner.setDriver(driver, proxy)`.
> 2. If don't open webdriver with proxy, how could you download files via proxy?

I will check that.Thank you.
